### PR TITLE
feat(github): support github app tokens

### LINKED ISSUE #6371: Refactor global.appMode to support app tokens
Currently `global.appMode` is used in a few places within the code to change behavior subtly if Renovate is running as a GitHub app. We should make this a first-class concept so that it's possible to supply an app-generated token for GitHub from the CLI.

### PR DESCRIPTION
**Pull request by rarkins**
Adds support for GitHub app tokens by recognizing the token prefix `x-access-token:`.

Closes #6371

### CODE REVIEW

File: lib/config/presets/github/index.ts [hunk 1]
```diff
@@ -15,9 +15,7 @@ export async function fetchJSONFile(
   const url = `${endpoint}repos/${repo}/contents/${fileName}`;
   const opts: HttpOptions = {
     headers: {
-      accept: global.appMode
-        ? 'application/vnd.github.machine-man-preview+json'
-        : 'application/vnd.github.v3+json',
+      accept: 'application/vnd.github.v3+json',
```
Comment (rarkins) [inline lib/config/presets/github/index.ts:18]: I'm thinking that accept headers could be better achieved using hostRules instead.

File: lib/platform/github/index.ts [hunk 2]
```diff
@@ -89,44 +91,50 @@ export async function initPlatform({
...
-          token,
-        }
-      )
-    ).body;
-    if (userEmail.length && userEmail[0].email) {
-      gitAuthor += ` <${userEmail[0].email}>`;
-    } else {
-      logger.debug('Cannot find an email address for Renovate user');
+  if (appMode) {
+    renovateUsername = username;
```
Comment (rarkins) [inline lib/platform/github/index.ts:95]: Need to throw error if username is not given

File: lib/platform/github/index.ts [hunk 3]
```diff
@@ -89,44 +91,50 @@ export async function initPlatform({
...
+    gitAuthor = `${username}@noreply.users.github.com`;
```
Comment (rarkins) [inline lib/platform/github/index.ts:96]: Need to customize for the host/endpoint

File: lib/config/presets/github/index.ts (hunk 1 above)
Comment (rarkins) [reply lib/config/presets/github/index.ts:20]: maybe remove this if already done in util/http

**Comment by viceice** [comment]
@rarkins let me know when i should review 🙃

File: lib/workers/repository/master-issue.ts [hunk 4]
```diff
@@ -193,7 +193,7 @@ export async function ensureMasterIssue(
   }
 
   // istanbul ignore if
-  if (global.appMode) {
+  if (config.appMode) {
```
Comment (rarkins) [inline lib/workers/repository/master-issue.ts:196]: Need to check if this is passed through

**Pull request closed without merging**